src/reducers/sources.js

```diff
@@ -69,14 +72,14 @@ function update(state = State(), action: Action) : Record<SourcesState> {
       });
 
     case "CLOSE_TAB":
-      availableTabs = removeSourceFromTabList(state.tabs, action.id);
+      availableTabs = removeSourceFromTabList(state, state.tabs, action.id);
```

Comment:
I think there is a way to lessen the workload. Instead of of loading the source text for every tabs, we only load the source text for the left most tab, so the other tabs would appear but their source text would not be loaded. 

If we do this, we would only call `dispatch(loadSourceText(source));` for the leftmost tab. Instead of calling `selectSource`, I would write a method to accommodate this. 

Is this ok?